### PR TITLE
Refactor CLI Commands

### PR DESCRIPTION
**Pull request by biniona-mongodb**
This PR does the following things:

- Separates "Actions" from "Command Modules" where an "Action" is bluehawk performing changes related to the output of the processor
- Removes yargs as a dependency from "Actions"

### CODE REVIEW

**Comment by biniona-mongodb** [comment]
@cbush  notes:

- CLI should now not export anything
- camel case "commandModules" folder
- rename "MainArgs" to "ActionArgs"

File: src/bluehawk/actions/actions.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export interface ActionArgs {
```
Comment (cbush) [inline src/bluehawk/actions/actions.ts:1]: Should this file be called "ActionArgs.ts" then?

File: src/cli/commandModules/index.ts [hunk 2]
```diff
@@ -1,5 +1,6 @@
 export * from "./check";
 export * from "./copy";
-export * from "./list/";
 export * from "./list";
+export * from "./listCommands";
+export * from "./listStates";
```
Comment (cbush) [inline src/cli/commandModules/index.ts:5]: Do you know if the list command is still working on the CLI?

The reason for the directory layout was that list.ts loads the subcommands subdirectory (just as the main cli loads the commandModules subdirectory). I think list.ts needs to be updated.

**Review by cbush — CHANGES_REQUESTED** [review]
Looks great! Couple issues to check.

File: src/cli/commandModules/index.ts (hunk 2 above)
Comment (biniona-mongodb) [reply src/cli/commandModules/index.ts:5]: The  list commands still work (just tested). All modules get loaded from the src/cli.ts `commandDir` call. I think the `commandDir` call in src/cli/commandModules/list did not do anything. I removed the builder method.

File: src/bluehawk/actions/actions.ts (hunk 1 above)
Comment (biniona-mongodb) [reply src/bluehawk/actions/actions.ts]: Renamed!

File: src/cli/commandModules/index.ts (hunk 2 above)
Comment (cbush) [reply src/cli/commandModules/index.ts:5]: The list command (`bluehawk list`) no longer lists the subcommands. This is what the commandDir call was doing.

Compare your change (above) with 0.6.0 (below):

image

Comment (biniona-mongodb) [reply src/cli/commandModules/index.ts]: Sorry, I now realize I changed the API and made `states` and `commands` no longer subcommands of `list`. Thanks for helping me understand the mechanism by which subcommands are created over slack.

I think I implemented the fix, let me know if you have any additional concerns.

**Review by cbush — APPROVED** [review]
Great work!

**Pull request merged**